### PR TITLE
🚧(dashboard) add consent models

### PR DESCRIPTION
**Pull request by ssorin**
## Added

- add consent models
- create core app
- add basic consent admin 
- create entity and delivery_point models
- add basic entity and delivery_point admin

### CODE REVIEW

File: src/dashboard/apps/core/models/entity.py
Comment (jmaupetit) [inline src/dashboard/apps/core/models/entity.py:1]: Having source files with a single model seems odd to me. Why not moving `DeliveryPoint` and `Entity` models in `core/models/__init__.py` instead?

File: src/dashboard/apps/core/models/entity.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+"""Dashboard core app models."""
+
+from django.db import models
+
+from apps.auth.models import DashboardUser as User
+from apps.core.models.abstract import DateBase
+
+
+class Entity(DateBase):
+    """Represents an operator in the system."""
```
Comment (jmaupetit) [inline src/dashboard/apps/core/models/entity.py:10]:
```suggestion
    """Represents an operator or an aggregator in the system."""
```

File: src/dashboard/apps/core/models/entity.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    name = models.CharField(max_length=64, unique=True)
+    user = models.ManyToManyField(User)
```
Comment (jmaupetit) [inline src/dashboard/apps/core/models/entity.py:13]:
```suggestion
    users = models.ManyToManyField(User)
```

File: src/dashboard/apps/core/models/entity.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    is_enabled = models.BooleanField(default=True)
```
Comment (jmaupetit) [inline src/dashboard/apps/core/models/entity.py:14]:
```suggestion
    is_active = models.BooleanField(default=True)
```

Comment (jmaupetit) [reply src/dashboard/apps/core/models/entity.py:14]: Do you think we need to inactivate entities?

File: src/dashboard/apps/core/models/delivery_point.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+from .abstract import DateBase
+from .entity import Entity
+
+
+class DeliveryPoint(DateBase):
+    """Represents a delivery point for electric vehicles."""
+
+    delivery_point_id = models.CharField(max_length=64)
+    entity = models.ManyToManyField(Entity)
```
Comment (jmaupetit) [inline src/dashboard/apps/core/models/delivery_point.py:14]:
```suggestion
    entities = models.ManyToManyField(Entity)
```

File: src/dashboard/apps/consent/models.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    CONSENT_STATUS_CHOICE = [
+        (AWAITING, _("Awaiting")),
+        (VALIDATED, _("Validated")),
+        (REVOKED, _("Revoked")),
+    ]
+
+    delivery_point = models.ForeignKey(DeliveryPoint, on_delete=models.CASCADE)
+    user = models.ForeignKey(
+        User, on_delete=models.SET_NULL, null=True, verbose_name=_("user")
+    )
```
Comment (jmaupetit) [inline src/dashboard/apps/consent/models.py:42]:
```suggestion
    created_by = models.ForeignKey(
        User, on_delete=models.SET_NULL, null=True, verbose_name=_("Created by")
    )
```

File: src/dashboard/apps/consent/models.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    status = models.CharField(
+        _("status"), max_length=20, choices=CONSENT_STATUS_CHOICE, default=AWAITING
+    )
+
+    # Validity period
+    validity_date_begin = models.DateTimeField(_("begin date"))
+    validity_date_expiration = models.DateTimeField(_("expiration date"))
```
Comment (jmaupetit) [inline src/dashboard/apps/consent/models.py:49]:
```suggestion
    start = models.DateTimeField(_("Start date"))
    end = models.DateTimeField(_("End date"))
```

File: src/dashboard/apps/consent/models.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    revoked_at = models.DateTimeField(_("revoked at"), null=True, blank=True)
+
+    class Meta:
+        """."""
```
Comment (jmaupetit) [inline src/dashboard/apps/consent/models.py:53]: FIXME

File: src/dashboard/Pipfile [hunk 8]
```diff
@@ -7,21 +7,23 @@ name = "pypi"
 Django = "==5.1.3"
 django-dsfr = "==1.4.3"
 django-environ = "==0.11.2"
+django-extensions = "==3.2.3"
```
[the remark this answers is not in the document]
Comment (ssorin) [reply src/dashboard/Pipfile:10]: yes, I also asked myself the question.
But `django-extension` offers features for `models` and `admin`...

File: src/dashboard/apps/consent/migrations/0002_remove_chargepointconsent_charge_point_and_more.py
[the remark this answers is not in the document]
Comment (ssorin) [reply src/dashboard/apps/consent/migrations/0002_remove_chargepointconsent_charge_point_and_more.py:1]: yes, I will reset the migrations when we are ok on all the `models` present here.

File: src/dashboard/apps/core/models/entity.py (hunk 3 above)
Comment (ssorin) [reply src/dashboard/apps/core/models/entity.py:14]: This is a concept present in the Carbure entity model

File: src/dashboard/apps/core/models/entity.py
Comment (ssorin) [reply src/dashboard/apps/core/models/entity.py:1]: I think this is a fairly common way to do things with Django.
Idon't really like models in the init: models are then "harder" to find.
I prefer to put all the models (`DeliveryPoint`, `Entity` and `DashboardBase`) in a single models.py

File: src/dashboard/Pipfile (hunk 8 above)
[the remark this answers is not in the document]
Comment (ssorin) [reply src/dashboard/Pipfile]: ✅

File: src/dashboard/apps/core/models/abstract.py [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    Abstract base class, will not be used to create any database table.
+    """
+
+    date_added = models.DateTimeField(
+        _("added date"), editable=False, default=timezone.now
+    )
+    date_updated = models.DateTimeField(_("date updated"), null=True, blank=True)
+
+    class Meta:
+        """."""
```
[the remark this answers is not in the document]
Comment (ssorin) [reply src/dashboard/apps/core/models/abstract.py:20]: ✅

File: src/dashboard/apps/consent/models.py (hunk 7 above)
Comment (ssorin) [reply src/dashboard/apps/consent/models.py:53]: ✅

File: src/dashboard/apps/core/models/delivery_point.py [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+    class Meta:
+        """."""
+
+        verbose_name = _("Delivery point")
+        verbose_name_plural = _("Delivery points")
+        ordering = ["delivery_point_id"]
+
+    def __str__(self):
+        """."""
```
[the remark this answers is not in the document]
Comment (ssorin) [reply src/dashboard/apps/core/models/delivery_point.py:25]: ✅

File: src/dashboard/apps/core/models/abstract.py [hunk 11]
```diff
@@ -0,0 +1,27 @@
+"""Dashboard core app models."""
+
+from django.db import models
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as _
+
+
+class DateBase(models.Model):
```
[the remark this answers is not in the document]
Comment (ssorin) [reply src/dashboard/apps/core/models/abstract.py:8]: ✅

File: src/dashboard/apps/core/models/abstract.py [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+    """Base model that includes date fields for tracking creation and updates.
+
+    Abstract base class, will not be used to create any database table.
+    """
+
```
[the remark this answers is not in the document]
Comment (ssorin) [reply src/dashboard/apps/core/models/abstract.py:13]: ✅

File: src/dashboard/apps/core/models/delivery_point.py [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+from django.utils.translation import gettext_lazy as _
+
+from .abstract import DateBase
+from .entity import Entity
+
+
+class DeliveryPoint(DateBase):
+    """Represents a delivery point for electric vehicles."""
+
+    delivery_point_id = models.CharField(max_length=64)
```
[the remark this answers is not in the document]
Comment (ssorin) [reply src/dashboard/apps/core/models/delivery_point.py:13]: ✅

File: src/dashboard/apps/core/models/delivery_point.py [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+    entity = models.ManyToManyField(Entity)
+    is_deleted = models.BooleanField(default=False)
```
[the remark this answers is not in the document]
Comment (ssorin) [reply src/dashboard/apps/core/models/delivery_point.py:15]: ✅

File: src/dashboard/apps/core/models/entity.py
Comment (ssorin) [reply src/dashboard/apps/core/models/entity.py:1]: ✅

File: src/dashboard/apps/core/models/entity.py (hunk 1 above)
Comment (ssorin) [reply src/dashboard/apps/core/models/entity.py:10]: or an aggregator

File: src/dashboard/apps/core/models/entity.py (hunk 2 above)
Comment (ssorin) [reply src/dashboard/apps/core/models/entity.py:13]: or an aggregator

File: src/dashboard/apps/core/models/delivery_point.py (hunk 4 above)
Comment (ssorin) [reply src/dashboard/apps/core/models/delivery_point.py:14]: ✅

File: src/dashboard/apps/core/models/entity.py (hunk 3 above)
Comment (ssorin) [reply src/dashboard/apps/core/models/entity.py:14]: I removed it

File: src/dashboard/apps/consent/models.py (hunk 5 above)
Comment (ssorin) [reply src/dashboard/apps/consent/models.py:42]: ✅

File: src/dashboard/apps/consent/models.py (hunk 6 above)
Comment (ssorin) [reply src/dashboard/apps/consent/models.py:49]: ✅

File: src/dashboard/apps/core/base.py
[the remark this answers is not in the document]
Comment (ssorin) [reply src/dashboard/apps/core/base.py:1]: I thought it was weird too 😅
Ok, so, I move everything in models.py

File: src/dashboard/apps/core/models.py [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+
+    Attributes:
+    - provider_id (CharField): stores the unique identifier for the delivery provider.
+    - entities (ManyToManyField): linking DeliveryPoint to associated Entity instances.
+    - is_active (BooleanField): indicating the active status of the delivery point.
+    """
+
+    provider_id = models.CharField(_("provider id"), max_length=64)
+    entities = models.ManyToManyField(Entity, verbose_name=_("entities"))
+    is_active = models.BooleanField(_("is active"), default=True)
```
[the remark this answers is not in the document]
Comment (ssorin) [reply src/dashboard/apps/core/models.py:42]: I usually put everything in lowercase:
- in the admin, Django takes care of putting the capital letter if necessary.
- when displaying a form submission error, the messages can be like "The field <field-name> is not valid". It is better that it is in lowercase at this level, to not have to manage the case.
- I have never seen an explicit recommendation for the verbose_name case, but I think that in the Django doc, in the code examples, verbose names are always in lowercase...

[the remark this answers is not in the document]
Comment (jmaupetit) [reply src/dashboard/apps/core/models.py:68]: Ok then! We need to ensure that this pattern is applied everywhere! :pray:

**Review by jmaupetit — APPROVED** [review]
Ship it!

**Pull request merged**